contracts/test/Summa.ts

```diff
@@ -346,9 +386,223 @@ describe("Summa Contract", () => {
+          cexAddress: defaultAbiCoder.encode(["address"], [account2.address]),
+          signature:
+            "0xb17a9e25265d3b88de7bfad81e7accad6e3d5612308ff83cc0fef76a34152b0444309e8fc3dea5139e49b6fc83a8553071a7af3d0cfd3fb8c1aea2a4c171729c1c",
+          message: message,
+        },
+      ];
+
+      const solvencyJson = fs.readFileSync(
+        path.resolve(
+          __dirname,
+          "../../zk_prover/examples/solvency_proof_solidity_calldata.json"
+        ),
+        "utf-8"
+      );
+      const solvencyCalldata: any = JSON.parse(solvencyJson);
+
+      mstRoot = solvencyCalldata.public_inputs[0];
+      solvencyProof = solvencyCalldata.proof;
+
+      const inclusionJson = fs.readFileSync(
+        path.resolve(
+          __dirname,
+          "../../zk_prover/examples/inclusion_proof_solidity_calldata.json"
+        ),
+        "utf-8"
+      );
+      const inclusionCalldata: any = JSON.parse(inclusionJson);
+
+      leafHash = inclusionCalldata.public_inputs[0];
+      mstRoot = inclusionCalldata.public_inputs[1];
+      inclusionProof = inclusionCalldata.proof;
+    });
+
+    it("should verify the proof of inclusion for the given public input", async () => {
+      await summa.submitProofOfAddressOwnership(ownedAddresses);
+      await summa.submitProofOfSolvency(
+        mstRoot,
+        [
+          {
+            chain: "ETH",
+            assetName: "ETH",
+            amount: BigNumber.from(556863),
+          },
+          {
+            chain: "ETH",
+            assetName: "USDT",
+            amount: BigNumber.from(556863),
+          },
+        ],
+        solvencyProof,
+        BigNumber.from(1693559255)
+      );
+      expect(
+        await summa.verifyInclusionProof(
+          inclusionProof,
+          [leafHash, mstRoot],
+          1693559255
+        )
+      ).to.be.equal(true);
+    });
+
+    it("should not verify with invalid MST root", async () => {
+      await summa.submitProofOfAddressOwnership(ownedAddresses);
+      await summa.submitProofOfSolvency(
+        mstRoot,
+        [
+          {
+            chain: "ETH",
+            assetName: "ETH",
+            amount: BigNumber.from(556863),
+          },
+          {
+            chain: "ETH",
+            assetName: "USDT",
+            amount: BigNumber.from(556863),
+          },
+        ],
+        solvencyProof,
+        BigNumber.from(1693559255)
+      );
+      mstRoot = BigNumber.from(0);
+      await expect(
+        summa.verifyInclusionProof(
+          inclusionProof,
+          [leafHash, mstRoot],
+          1693559255
+        )
+      ).to.be.revertedWith("Invalid MST root");
+    });
+
+    it("should not verify if the MST root lookup by timestamp returns an incorrect MST root", async () => {
+      // The lookup will return a zero MST root as no MST root has been stored yet
+      await expect(
+        summa.verifyInclusionProof(
+          inclusionProof,
+          [leafHash, mstRoot],
+          1693559255
```

Comment:
If we use a different timestamp here, for example `1693559256`, we could move the code lines that generate transactions with `await summa.submitProofOfAddressOwnership` and `await summa.submitProofOfSolvency` to a `beforeEach` block. 

This would streamline the setup for all inclusion tests.

Reply:
I get the idea of extracting the functions but I don't get how it's related to a particular value of the timestamp.

Reply:
What I meant is that querying `mstRoots` with a timestamp that hasn't been set will return 0 for the `mstRoot`. 
However, it looks like you've already taken care of that in the code, so we're good to go.